mutt/string.c

```diff
@@ -255,7 +255,13 @@ char *mutt_str_dup(const char *str)
   if (!str || (*str == '\0'))
     return NULL;
 
-  return strdup(str);
+  char *p = strdup(str);
+  if (!p)
+  {
+    mutt_error(_("Out of memory")); // LCOV_EXCL_LINE
+    mutt_exit(1);                   // LCOV_EXCL_LINE
```

Comment:
Maybe we should use

```
mutt_error("%s", strerror(errno));
```

instead of having our own string.

Reply:
@flatcap , I copied that from

```c
$ grepc mutt_mem_calloc .
./mutt/memory.c:void *mutt_mem_calloc(size_t nmemb, size_t size)
{
  if ((nmemb == 0) || (size == 0))
    return NULL;

  void *p = calloc(nmemb, size);
  if (!p)
  {
    mutt_error(_("Out of memory")); // LCOV_EXCL_LINE
    mutt_exit(1);                   // LCOV_EXCL_LINE
  }
  return p;
}
./mutt/memory.h:void *mutt_mem_calloc(size_t nmemb, size_t size);
```

Should I fix both to use *strerror*(3)?  That would reduce localization effort.

Reply:
> Should I fix both to use _strerror_(3)?

Good idea.  Yes, please.

FYI.
Upstream have taken an interesting route.
Worried about re-entrancy during the apocalyse, they avoid `gettext()` and `printf()`.
They save the translation on startup and use `write(2)` to display it.

Reply:
Could you please point to specific commits of upstream?  I'd like to see exactly what they did.

Reply:
A quick grep shows:

Reply:
Thanks!  It looks a bit complex... I'll check some day.  Maybe we could track it in a bug report.